app/lib/activity_stream_reader.rb

```diff
@@ -56,6 +56,7 @@ def find_by_id(item)
     elsif metadata_source == "aspace"
       source_id = ("/" + source_id_type + "/" + source_id).to_s
       oid = ParentObject.where("aspace_uri" => source_id.to_s)&.first&.oid
+      oid = DependentObject.where(dependent_uri: source_id.to_s)&.first&.parent_object_id&.to_s if oid.nil?
```

Comment:
this feels like maybe a method that's missing. that's a _lot_ of nil checking in the middle of doing find_by_id. (6 in 2 lines by my count). Maybe extracting it out to a `oid_from_source_id(source_id)` method could make that more readable / reusable?

Reply:
It's possible (and likely) that multiple aspace objects may depend on one entry in the activity stream.
So, just getting the first one will not find all the oids that need to be updated in some cases.